### DISCUSSION TITLE
getStaticPaths with Firestore subCollections?

### DISCUSSION THREAD

**Discussion by nonoumasy** [vercel/next.js#25623] [Help]
I'm still new to NextJS and still unclear on some details on dynamic routes.
I have a site that has Collections of books. 
https://babel.vercel.app/ 

bookCollection is a Firestore parent collection and
book is a subcollection of bookCollection

How does one get the paths for a nested dynamic routes like this, specifically the subcollection 'books'?
Anyone use getStaticPaths with Firestore subcollections?

This is currently how I'm implementing this.

```
//get all the paths for books
export const getStaticPaths = async () => {
    const res = await firestore.collectionGroup("books").get()
    const paths = res.docs.map(item => {
        return {
        params: { bookId: item?.id }
        }
    })
    return {
        paths,
        fallback: true,
    }
}

export const getStaticProps = async ({params}) => {
    const data = await getBook(params.bookId)
    return {
        props: {data}
    }
}
export default function BookPage({ data }) {
 //render the page here
 return ( ... )
}
```

**Comment by leerob** [comment]
Your approach looks correct – fetch all of the books, iterate over the list, and return an array of paths to `getStaticPaths`. 

[Here's a Next.js example using Firebase](https://github.com/leerob/fastfeedback/blob/master/pages/embed/%5B...site%5D.js#L24-L36) which should help.

**Reply by nonoumasy** [reply]
thanks @leerob . 
I'm still having this issue which that I cannot get the parent collection id using context using getStaticPaths. The only way I solved this issue is by using getServerSideProps. I don't get the benefit of SSG.

```
export async function getServerSideProps(context) {
  const { id: parentCollectionId } = context.params
  const { childCollectionId } = context.query

  const someData = await getEntry(parentCollectionId, childCollectionId)
  const data = JSON.stringify(someData)

  return {
    props: {
      data,
    },
  }
}
```
